sql/handler.cc

```diff
@@ -3857,19 +3858,22 @@ int handler::ha_index_read_idx_map(uchar *buf, uint index, const uchar *key,
   DBUG_ASSERT(end_range == NULL);
   TABLE_IO_WAIT(tracker, PSI_TABLE_FETCH_ROW, index, result,
     { result= index_read_idx_map(buf, index, key, keypart_map, find_flag); })
-  increment_statistics(&SSV::ha_read_key_count);
-  if (!result)
+  if (update)
   {
-    rows_stats.key_read_hit++;
-    update_rows_read();
-    index_rows_read[index]++;
-    if (table->vfield && buf == table->record[0])
-      table->update_virtual_fields(this, VCOL_UPDATE_FOR_READ);
-  }
-  else
-  {
-    status_var_increment(table->in_use->status_var.ha_read_key_miss);
-    rows_stats.key_read_miss++;
+    increment_statistics(&SSV::ha_read_key_count);
```

Comment:
No, sorry, I'm afraid this won't do. If `index_read_idx_map` is called, the counter of `index_read_idx_map` calls should be incremented. This is the whole purpose of this counter, you cannot just arbitrarily skip it.

Same for everything below.

Reply:
I have just changed.
However, this feature implementation will introduce failure in unit test, 'main.limit_rows_examined', when invokes the procedure, 'test_limit_rows'. Therefore, I added the flag, 'update', to not update the operation, LIMIT ROWS EXAMINED.